### PR TITLE
Code Smells (use equals, remove arrays from parameters, wild card returns, mergeable if statements)

### PR DESCRIPTION
**Pull request by capricornunicorn123**
### Prerequisites

- [ ] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

Changes address the following Sonar Rules

**(Java) "==" and "!=" should not be used when "equals" is overridden** - The following case could not be addressed

- ExpressionParser (au.gov.asd.tac.constellation.graph.value.expression)

**(Java) Method parameters should be declared with base types** - All cases addressed

**(Java) Generic wildcard types should not be used in return types** - All cases could not be addressed.
Refactoring wild cards in all cases will result in a knock-on effect for a large number of classes across many modules. If warranted, this work can be undertaken but does not seem a priority given the effort vs value.

**(Java) Mergeable "if" statements should be combined** - All cases addressed

### Applicable Issues
#2022

### CODE REVIEW

File: CoreTableView/src/au/gov/asd/tac/constellation/views/tableview/factory/TableViewPageFactory.java [hunk 1]
```diff
@@ -117,11 +117,11 @@ public Node call(Integer pageIndex) {
         if (allTableRows != null) {
 
             // This action only effects selected only mode variables but runs no
-            // matter if the table is in selected only mode or not.
+            // matter if the table is in selected only mode or not.pluginExecution(Java) "==" and "!=" should not be used when "equals" is overridden
```
Comment (antares1470) [inline CoreTableView/src/au/gov/asd/tac/constellation/views/tableview/factory/TableViewPageFactory.java:120]: the addition here could probably be removed

File: CoreUtilities/src/au/gov/asd/tac/constellation/utilities/text/StringUtilities.java [hunk 2]
```diff
@@ -38,6 +39,14 @@ public class StringUtilities {
     private static final Pattern OPENING_SQUARE_BRACKET = Pattern.compile("^\\[");
     private static final Pattern ENDING_SQUARE_BRACKET = Pattern.compile("]$");
 
+    public static boolean endsWithAny(String name, String... endings) {
```
Comment (antares1470) [inline CoreUtilities/src/au/gov/asd/tac/constellation/utilities/text/StringUtilities.java:42]: parameters final

File: CoreUtilities/src/au/gov/asd/tac/constellation/utilities/text/StringUtilities.java [hunk 3]
```diff
@@ -38,6 +39,14 @@ public class StringUtilities {
...
+        return StringUtils.endsWithAny(name, endings);
+    }
+
+    public static boolean equalsAny(String attributeType, String... values) {
```
Comment (antares1470) [inline CoreUtilities/src/au/gov/asd/tac/constellation/utilities/text/StringUtilities.java:46]: same here

**Review by antares1470 — CHANGES_REQUESTED** [review]

File: CoreTableView/src/au/gov/asd/tac/constellation/views/tableview/factory/TableViewPageFactory.java (hunk 1 above)
Comment (capricornunicorn123) [reply CoreTableView/src/au/gov/asd/tac/constellation/views/tableview/factory/TableViewPageFactory.java:120]: I think this was a rogue CTRL+V thanks for catching

**Review by Delphinus8821 — APPROVED** [review]
Looks good once requested changes have been done.

**Review by antares1470 — APPROVED** [review]

**Pull request merged**